internal/app/plan.go

```diff
@@ -86,25 +86,57 @@ func (p *plan) exec() {
 		log.Info("Nothing to execute")
 	}
 
+	wg := sync.WaitGroup{}
+	sem := make(chan struct{}, flags.parallel)
+	var fail bool
+	var priorities []int
+	pl := make(map[int][]orderedCommand)
 	for _, cmd := range p.Commands {
-		log.Notice(cmd.Command.Description)
-		result := cmd.Command.exec()
-		if cmd.targetRelease != nil && !flags.dryRun && !flags.destroy {
-			cmd.targetRelease.label()
+		priorities = append(priorities, cmd.Priority)
+		pl[cmd.Priority] = append(pl[cmd.Priority], cmd)
+	}
+
+	for _, priority := range priorities {
+		c := make(chan string, len(pl[priority]))
+		for _, cmd := range pl[priority] {
+			sem <- struct{}{}
+			wg.Add(1)
+			go func(cmd orderedCommand) {
+				defer func() {
+					wg.Done()
+					<-sem
+				}()
+				log.Notice(cmd.Command.Description)
+				result := cmd.Command.exec()
+				if cmd.targetRelease != nil && !flags.dryRun && !flags.destroy {
+					cmd.targetRelease.label()
+				}
+				if result.code != 0 {
+					errorMsg := result.errors
```

Comment:
we should add chart name to the error message since for multiple errors this becomes unclear

Reply:
I've updated the message a few lines below.